File: common_business/src/main/java/com/xiaoyv/busines/room/dao/RdpDao.java
package com.xiaoyv.busines.room.dao;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import com.xiaoyv.busines.room.entity.RdpEntity;

import java.util.List;

/**
 * rdpDao
 *
 * @author why
 * @since 2020/11/29
 **/
@Dao
public interface RdpDao {
    @Query("SELECT * FROM rdp")
    List<RdpEntity> getAll();

    @Query("SELECT * FROM rdp WHERE id =:id LIMIT 1")
    RdpEntity getById(long id);

    @Query("SELECT * FROM rdp WHERE ip IN (:ip)")
    List<RdpEntity> getAllByIps(int... ip);

    @Query("SELECT * FROM rdp WHERE `group` IN (:group)")
    List<RdpEntity> getAllByGroup(String... group);

    @Query("SELECT * FROM rdp WHERE label LIKE :label")
    List<RdpEntity> findAllByLabel(String label);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insert(RdpEntity... rdpEntities);

    @Update
    void update(RdpEntity... rdpEntities);

    @Delete
    void delete(RdpEntity... rdpEntity);
}

File: common_business/src/main/java/com/xiaoyv/busines/room/dao/RdpDao.kt
package com.xiaoyv.busines.room.dao

import androidx.room.*
import com.xiaoyv.busines.room.entity.RdpEntity

/**
 * rdpDao
 *
 * @author why
 * @since 2020/11/29
 */
@Dao
interface RdpDao {
    @Query("SELECT * FROM rdp")
    fun queryAll(): List<RdpEntity>

    @Query("SELECT * FROM rdp WHERE id =:id LIMIT 1")
    fun queryById(id: Long): RdpEntity?

    @Query("SELECT * FROM rdp WHERE ip IN (:ip)")
    fun queryByIps(vararg ip: Int): List<RdpEntity>

    @Query("SELECT * FROM rdp WHERE label LIKE :label")
    fun queryByLabel(label: String): List<RdpEntity>

    @Query("SELECT * FROM rdp WHERE `group` IN (:group)")
    fun queryByGroup(vararg group: String): List<RdpEntity>

    @Query("SELECT DISTINCT `group` FROM rdp ORDER BY `group` COLLATE LOCALIZED ASC")
    fun queryGroup(): List<String>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(vararg rdpEntities: RdpEntity)

    @Update
    fun update(vararg rdpEntities: RdpEntity)

    @Delete
    fun delete(vararg rdpEntity: RdpEntity)
}